### PR TITLE
CustomGlobalExceptionHandler, validation in Dto

### PR DESCRIPTION
**Pull request by DEz-1996**
Add CustomGlobalExceptionHandler class
Add validation for CreateBookRequestDto

### CODE REVIEW

File: src/main/java/online/book/store/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class CreateBookRequestDto {
-    private String title;
+    @NotNull
+    @Size(min = 5, message = "author size must be min 5")
```
Comment (AgroFix) [inline src/main/java/online/book/store/dto/CreateBookRequestDto.java]: You can also add the maximum number of characters, which is set in the field property in the database. If I'm not mistaken default is 255 characters

File: src/main/java/online/book/store/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
     private String author;
+    @NotNull
+    @Size(min = 5, message = "title size must be min 5")
```
Comment (AgroFix) [inline src/main/java/online/book/store/dto/CreateBookRequestDto.java]: Also can add max number of characters

File: src/main/java/online/book/store/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
+    private String title;
+    @NotNull
+    @Size(min = 10, max = 13)
```
Comment (AgroFix) [inline src/main/java/online/book/store/dto/CreateBookRequestDto.java]: If I am right, we will be able to enter 11 or 12 characters, which is not correct

**Review by AgroFix — APPROVED** [review]

File: src/main/java/online/book/store/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,15 +1,27 @@
...
 public class CreateBookRequestDto {
-    private String title;
+    @NotNull
+    @Size(min = 5, max = 255)
     private String author;
+    @NotNull
+    @Size(min = 5, max = 255)
+    private String title;
+    @NotNull
+    @Pattern(regexp = "\\d{10}|\\d{13}", message = "must be 10 or 13")
```
Comment (bhdnchui) [inline src/main/java/online/book/store/dto/CreateBookRequestDto.java]: Minor: It would be great to tell user what 10 and 13 are. It is length =)
```suggestion
    @Pattern(regexp = "\\d{10}|\\d{13}", message = "length must be 10 or 13 chars")
```

File: src/main/java/online/book/store/repository/BookRepository.java [hunk 5]
```diff
@@ -2,8 +2,6 @@
 
 import online.book.store.model.Book;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.stereotype.Repository;
 
-@Repository
```
Comment (bhdnchui) [inline src/main/java/online/book/store/repository/BookRepository.java:7]: Why did you remove this annotation?

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Just a few points to consider.

File: src/main/java/online/book/store/repository/BookRepository.java (hunk 5 above)
Comment (DEz-1996) [reply src/main/java/online/book/store/repository/BookRepository.java:7]: My bad!
I thought that I added this change in the previous PR before the merge.
Because when we use JpaRepository interface, then annotation @Repository is not necessary
Approved by mentor's comment from previous PR:
"It's not necessary for interfaces which extend Springs JPA related interfaces like JpaRepository and Spring applies it by default."

**Review by kshuryhin — APPROVED** [review]
looks good
